src/languages/internal/languagesservice.cpp

```diff
@@ -215,6 +215,22 @@ QString LanguagesService::effectiveLanguageCode(const QString& languageCode) con
                 return code;
             }
 
+            // For some languages, we only have long, specific codes in LanguagesHash / languages.json.
+            // If the user's system language is just "en", or "en_CY" for example, this will not be found.
+            // So, for short codes that are at risk, we map them to long codes that we do have.
+            // Should be kept in sync with languages.json.
+            static const std::map<QString, QString> SHORT_TO_LONG_CODES {
```

Comment:
I think we can move it to configuration
Like fallbacksForLanguages

Reply:
I prefer not moving it to configuration, because this relates quite closely to this `tryCode` algorithm, and actually all logic about which languages we support happens in LanguagesService. I hope this is okay :)